grype/db/v5/namespace/index.go

```diff
@@ -12,6 +13,8 @@ import (
 	syftPkg "github.com/anchore/syft/syft/pkg"
 )
 
+var alpineVersionRegularExpression = regexp.MustCompile(`^(\d+)\.(\d+)(?:\.(\d+))?(?:-r(\d+))?$`)
+
```

Comment:
For **lines 100-103** _(GitHub isn't letting me comment on those lines for some reason, sorry!)_:

If the Alpine version string is something like `3.16.0-r4`, we'll fall into the conditional case in these lines and potentially return the wrong namespace. The version string `3.16.0-r4` should map to `alpine:edge`, but instead the code will see that `len(versionSegments) == 3` and that the `alpine:3.16` namespace is present in the index, so I believe it will incorrectly return `alpine:3.16` in that scenario.

Reply:
It might be useful to add a test case to catch this scenario